### PR TITLE
sequtils: simplify `toSeq` implementation

### PR DESCRIPTION
**Pull request by zerbina**
## Summary

Don't use complex macros, but instead use `compiles` + generic routines
for achieving the same. This should slightly improve compile times for
code often using `toSeq`, and is also another step towards less macro
usage during bootstrapping.

## Details

This change partially reverts https://github.com/nim-works/nimskull/pull/151,
which replaced the usage of `compiles` with macros. However, compared to
macros, `compiles` is the less problematic feature usage here, allowing
the implementation to be significantly less complex. In addition, the
`compiles` here is also easier to replace with type-inspection routines
in the future (such as `x is iterator`, once that works).

### CODE REVIEW

File: lib/pure/collections/sequtils.nim [hunk 1]
```diff
@@ -844,10 +784,16 @@ macro toSeq*(arg: untyped): untyped =
     assert mySeq1 == @[1, 2, 3, 4, 5]
     assert mySeq2 == @[1'i8, 3, 5]
 
-  if arg.kind in {nnkSym, nnkIdent, nnkOpenSymChoice, nnkClosedSymChoice}:
-    result = newCall(bindSym"toSeqImplSingleSym", arg)
+  mixin items
+  when compiles(typeof(items(arg))):
+    # arg is something that supports the ``items`` iterator:
+    toSeqBuiltin[typeof(items(arg))](arg)
+  elif compiles(typeof(arg())):
```
Comment (zerbina) [inline lib/pure/collections/sequtils.nim:791]: This should be turned into `arg is iterator`, once the compiler support testing for iterator types that.

**Review by saem — APPROVED** [review]
lesson learned... hopefully. 🤣 

Type driven is better than AST driven... especially as you said once we have the ability to test for `iterator`.

**Comment by saem** [comment]
/merge

looks like a spurious failure

**Pull request merged**